src/SavedSearch.php

```diff
@@ -561,6 +561,9 @@ public function prepareQueryToUse($type, $query_tab, $enable_partial_warnings =
                     unset($query_tab['criteria']);
                     $new_key = 0;
                     foreach ($query_tab_save['criteria'] as $key => $val) {
+                        if (isset($val['itemtype']) && isset($val['meta'])) {
+                            $opt = Search::getCleanedOptions($val['itemtype']);
+                        }
```

Comment:
I think it may fail if, ater the meta, there is another criteria related to main item. Indeed, in this case, `$opt` will have been overriden and will not correspond to main item.
To not introduce performances issues here, following logic should probably implemented
```php
// on line 555
$itemtype_so = [
    $this->fields['itemtype'] => Search::getCleanedOptions($this->fields['itemtype']);
];

// in the `foreach ($query_tab_save['criteria'] as $key => $val) {` loop
$opt = [];
if (!isset($val['meta'])) {
    $opt = $itemtype_so[$this->fields['itemtype']];
} elseif (isset($val['itemtype'])) {
    if (!array_key_exists($val['itemtype'], $itemtype_so)) {
        $itemtype_so[$val['itemtype']] = Search::getCleanedOptions($val['itemtype']);
    }
    $opt = $itemtype_so[$val['itemtype']];
}
```

Reply:
done